examples/server/server.cpp

```diff
@@ -3106,63 +3122,56 @@ int main(int argc, char ** argv) {
         return httplib::Server::HandlerResponse::Unhandled;
     });
 
-    //
-    // Route handlers (or controllers)
-    //
+    auto middleware_model_loading = [&sparams, &res_error](const httplib::Request & req, httplib::Response & res, server_state current_state) {
+        static const std::set<std::string> invalid_endpoints = {
+            "/slots",
+            "/metrics",
+            "/props",
+            "/v1/models",
+            "/completion",
+            "/completions",
+            "/v1/completions",
+            "/chat/completions",
+            "/v1/chat/completions",
+            "/infill",
+            "/tokenize",
+            "/detokenize",
+            "/embedding",
+            "/embeddings",
+            "/v1/embeddings",
+        };
 
-    const auto handle_health = [&](const httplib::Request & req, httplib::Response & res) {
-        server_state current_state = state.load();
+        // If path is not in invalid_endpoints list, skip validation
+        if (invalid_endpoints.find(req.path) == invalid_endpoints.end()) {
```

Comment:
Here can we simply check if `req.path != "/health"` instead of listing all the blocked endpoints?

Reply:
Indeed but I guess that would mean answer 503 also for the static pages, is that ok?

Also found that GH actions is using `/v1/health` but it's not enabled anywhere from what I can see:
https://github.com/ggerganov/llama.cpp/blob/b9613ef11a748ba3b8961fd0504aca89134d705e/examples/server/server.cpp#L2864

Reply:
For simplification, we can return 503 for static page if the server is not yet ready. Currently, frontend does not know what to do when server is not yet ready (logically it should show an error message), so I think for now we don't need to care about that.

And yes you can add `/v1/health`. It seem to be an OAI-compat endpoint.

Reply:
added `&params` as well